src/SingleBookingPage/BookingDetail.js

```diff
@@ -54,21 +54,53 @@ class BookingDetail extends React.Component<Props> {
     return date ? new Date(date) : null;
   };
 
-  render() {
-    const { booking } = this.props;
-    const departure = this.getDepartureByType(booking);
-    const timeDelta = departure
-      ? DateTime.fromJSDate(departure, { zone: 'utc' }).diffNow('hours').hours
+  getArrivalByType = (booking: NearestBooking_booking) => {
+    let date = null;
+
+    if (booking.type === bookingTypes.ONE_WAY) {
+      date = idx(booking, _ => _.trip.arrival.time);
+    }
+
+    if (booking.type === bookingTypes.RETURN) {
+      date = idx(booking, _ => _.inbound.arrival.time);
+    }
+
+    if (booking.type === bookingTypes.MULTICITY) {
+      date = idx(booking, _ => _.end.time);
+    }
+
+    return date ? new Date(date) : null;
+  };
+
+  decideIfIsFutureAndUrgent = (time: ?Date) => {
+    const timeDelta = time
+      ? DateTime.fromJSDate(time, { zone: 'utc' }).diffNow('hours').hours
       : null;
     const isUrgent =
       timeDelta !== null && URGENCY_THRESHOLD > timeDelta && timeDelta >= 0;
-    // show notification only when the whole trip doesn't started yet
-    const showNotification = timeDelta !== null && timeDelta > 0;
+    return {
+      timeDelta,
+      isFuture: timeDelta !== null && timeDelta > 0,
+      isUrgent,
+    };
+  };
+
+  render() {
+    const { booking } = this.props;
+    const departureTime = this.getDepartureByType(booking);
+    const arrivalTime = this.getArrivalByType(booking);
+    const departureTimeDelta = this.decideIfIsFutureAndUrgent(departureTime);
```

Comment:
Not life-changing, but I believe the `departureTimeDelta` variable name is confusing. What about something like `departureInfo`?